lib/constructs/provisioning-sfn-workflow.test.ts

```diff
@@ -0,0 +1,88 @@
+import * as cdk from "aws-cdk-lib";
+import { Match, Template } from "aws-cdk-lib/assertions";
+import { ProvisioningWorkflow } from "./provisioning-sfn-workflow";
+import { CfnRole } from "aws-cdk-lib/aws-iam";
+
+describe("Provisioning Workflow Tests", () => {
+  let stack: cdk.Stack;
+  let template: Template;
+  let provisioningSfn: ProvisioningWorkflow;
+  let queuePolicyDocument: Record<string, unknown>;
+  let allowedRoles: Array<CfnRole>;
+
+  // Create the stack once and then run different tests against it
+  beforeAll(() => {
+    // GIVEN
+    const app = new cdk.App();
+    stack = new cdk.Stack(app, "TestStack");
+    // WHEN
+    provisioningSfn = new ProvisioningWorkflow(stack, "TestWorkflow", { environmentName: "testEnv" });
+    template = Template.fromStack(stack);
+
+    // Define Fixtures
+    queuePolicyDocument = {
+      Statement: [
+        {
+          Action: ["sqs:SendMessage", "sqs:GetQueueAttributes", "sqs:GetQueueUrl"],
+          Effect: "Allow",
+          Resource: {
+            "Fn::GetAtt": [
+              stack.resolve((provisioningSfn.provisioningJobsQueue.node.defaultChild as cdk.CfnResource).logicalId),
+              "Arn",
+            ],
+          },
```

Comment:
Idk how much cleaner it'd be to do this since it'd also require an import of `import * as sqs from "aws-cdk-lib/aws-sqs"`

```suggestion
          Resource: stack.resolve((provisioningSfn.provisioningJobsQueue.defaultChild as sqs.CfnQueue).attrArn),
```

Reply:
Definitely a bit cleaner - added the suggestion and the test still passes.